### ISSUE TITLE
Latex base template: change \usepackage[utf8x]{inputenc} to \usepackage[utf8]{inputenc}

### ISSUE THREAD

**Issue by alcrene** [jupyter/nbconvert#530]
The general consensus within the TeX community is that
`\usepackage[utf8]{inputenc}`
is preferable to using the `[utf8x]` option, which conflicts with a number of other packages. See for example http://tex.stackexchange.com/q/13067.

In my case it conflicts with my DeclareUnicodeChar definitions, which I load from a common file.

Not wanting to fork templates/latex/base.tplx for such a simple change, I added a bit of `\makeatletter` hackery to my template to force `inputenc` to load with the `utf8` option, but I think it would generally be nicer for people looking to extend the default templates if this option was used by default.

**Comment by michaelpacer** [comment]
I want to remove both of these, and ideally including mathspec rather than fontspec. See #537 for a more comprehensive PR trying to improve our unicode support.

Developing tests for LaTeX templates is an ongoing challenge.

**Comment by jochym** [comment]
As of now the unicode in pdf converter is broken. A lot of characters are simply missing in the output. This is caused by xelatex back-end. If you convert to latex and run pdflatex all is fine. I can provide simple example if needed.

**Comment by alcrene** [comment]
Normally xelatex provides much better character support than pdflatex, but you may be seeing issues because the current preamble is still using a few pdflatex packages (the ones I mentioned in this issue).

Perhaps as a warning to future readers: the workaround of using pdflatex is likely to break at some point in the future, when the latex template is updated with xetex-only replacements packages. At that point though, it should be possible to compile directly to a pdf, or convert to latex and run xelatex.

**Comment by jochym** [comment]
The problem is the template is not coordinated with the nbconvert setup - and this is sub-optimal. Xetex is better as a back-and maybe we can make quick update to latex templates to fix this problem? It is slightly annoying to have half of your name removed ... (my co-author)

**Comment by mpacer** [comment]
Yes, I've been intending to get back to this, but I was having trouble reproducing any (marginally) successful solutions across systems (especially on travis for testing. 

This seems to be (in part) related to font support: latin modern is better than computer modern for Unicode purposes, but it's still pretty impoverished (e.g., doesn't seem to have greek characters).

If you want to see the earlier efforts: https://github.com/jupyter/nbconvert/pull/537 

But I'm extremely sorry that that is happening to your co-author. My hope is to have a solution soon.

**Comment by alcrene** [comment]
As an immediate workaround, you can derive the latex template locally and change the packages to something that works on your system. I use something like the following:

    ((*- extends 'article.tplx' -*))
    [unrelated template changes]
    ((* block packages *))
        % Hide [utf8x]{inputenc} as it should not be used with xetex.
        %Also hide ucs which conflicts with a bunch of stuff
        % http://tex.stackexchange.com/a/39418
        \makeatletter
        \newcommand{\dontusepackage}[2][]{%
          \@namedef{ver@#2.sty}{9999/12/31}%
          \@namedef{opt@#2.sty}{#1}}
        \makeatother
        \dontusepackage[utf8x]{inputenc}
        \dontusepackage[mathletters]{ucs}
    ((( super() )))
        \usepackage{unicode-math}
        \usepackage{fontspec}
        \usepackage[Latin,Greek]{ucharclasses}
        \newfontfamily\substitutefont{CMU Serif}
        \setTransitionsForGreek{\begingroup\substitutefont}{\endgroup}
    ((* endblock packages *))

Save that as `mylatex.tplx`, put it in the same directory as your notebook (or your global nbconvert templates directory), and export the notebook with

    jupyter nbconvert --to pdf [notebook.ipynb] --template=mylatex.tplx

A few caveats:
1. As mspacer has said, finding a solution that works across systems is difficult, so you may have to tweak this a bit. It works for me on Ubuntu and openSUSE.
2. I wrote some of that from memory, so there may be typos.

**Comment by jochym** [comment]
Thanks. That i what I plan to do. The code above will help. But still including thin in studenta materials is sub-optimal from my POV. Anyway - thanks for help. I hope a permanent solution is not far off.
